db/crud.go

```diff
@@ -827,19 +825,53 @@ func (db *Database) PutExistingRev(docid string, body Body, docHistory []string,
 
 		// Process the attachments, replacing bodies with digests.
 		parentRevID := doc.History[newRev].Parent
-		newAttachments, err := db.storeAttachments(doc, body, generation, parentRevID, docHistory)
+		newAttachments, err := db.storeAttachmentsDoc(doc, newDoc.DocAttachments, generation, parentRevID, docHistory)
 		if err != nil {
 			return nil, nil, nil, err
 		}
-		// move _attachment metadata to syncdata of doc
-		doc.SyncData.Attachments = GetBodyAttachments(body)
-		delete(body, BodyAttachments)
 
-		body[BodyRev] = newRev
-		return body, newAttachments, nil, nil
+		doc.SyncData.Attachments = newDoc.DocAttachments
+		newDoc.RevID = newRev
+
+		return newDoc, newAttachments, nil, nil
 	})
 
+	return doc, newRev, err
+}
+
+func (db *Database) PutExistingRevREST(docid string, body Body, docHistory []string, noConflicts bool) (doc *Document, err error) {
+
+	// Blip pulls out revid, id, attachments, expiry
+
+	expiry, _ := body.ExtractExpiry()
```

Comment:
As discussed, could use a consistent 'Extract...' approach for all named properties (to retrieve and delete).